### ISSUE TITLE
Feature Request: Initialization of components by string

### ISSUE THREAD

**Issue by mwbrulhardt** [notadamking/tensortrade#33]
# Explanation
The purpose of this feature is to make commonly used components accessible by string in the context of initializing trading environments or trading strategies. This is similar in spirit to compiling models  in keras by setting loss and optimizers to be a string such as:
```python
model.compile(loss='mse', optimizer='adam')
```
This will make it easy for people to start using TensorTrade quickly and will also help more advanced users perform fast experimentation without having to do a lot of imports for common functionalities.

# Example
One example would be the following:
```python
from tensortrade.exchanges.simulated import FBMExchange
from tensortrade.rewards import SimpleProfitStrategy
from tensortrade.actions import DiscreteActionStrategy
from tensortrade.environments import TradingEnvironment

exchange = FBMExchange(base_instrument='BTC', timeframe='1h')
reward_strategy = SimpleProfitStrategy()
action_strategy = DiscreteActionStrategy(n_actions=20, instrument_symbol='USD/BTC')

environment = TradingEnvironment(exchange=exchange,
                                 action_strategy=action_strategy,
                                 reward_strategy=reward_strategy)
```

This would be a common configuration to run and with string initialization it would look like the following:
```python
from tensortrade.environments import TradingEnvironment

environment = TradingEnvironment(exchange='fbm', 
                                 action_strategy='discrete', 
                                 reward_strategy='simple')
```

Exchanges will not always be so simple so this example is good for getting people started with tensortrade. However, a more common use case would be making a complex exchange configuration, and then initializing the actions and rewards using strings:
```python
from tensortrade.exchanges.simulated import FBMExchange
from tensortrade.rewards import SimpleProfitStrategy
from tensortrade.actions import DiscreteActionStrategy
from tensortrade.environments import TradingEnvironment

exchange = FBMExchange(base_instrument='BTC', timeframe='1h')

environment = TradingEnvironment(exchange=exchange,
                                 action_strategy='discrete',
                                 reward_strategy='simple')
```
I already have a pull request with the implementation and would want to upgrade the tutorials to reflect this updated method of initialization.

**Comment by notadamking** [comment]
Thanks Matthew for raising this issue. I am going to add some Gitcoin funding for the completion of this feature.

**Funding:**
0.1 ETH (~$20 USD)

**Description:**
Allow each TensorTrade component to be initialized via string.

**Requirements:**
* Each TensorTrade component should be able to be initialized via Class object or string, using an accompanying `get` method. (E.g. `actions.get('SimpleDiscrete')`).
* Each TensorTrade component that takes another component as a parameter (e.g. `TradingEnvironment`, `InstrumentExchange`, etc.) should accept either a component instance or a `str`.

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. Submitted 4. **Done** 

<hr>

__This issue now has a funding of 0.1 ETH (18.3 USD @ $182.98/ETH)  attached to it as part of the tensortrade-org fund.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/notadamking/tensortrade/33/3548).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/notadamking/tensortrade/33/3548).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $155,888.34 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 1 month from now.
Please review their action plans below:


**1) [mwbrulhardt](https://gitcoin.co/profile/mwbrulhardt) has started work.**

1. Create unit tests for each component to be initialized by string.
2. Write get methods to call each one and pass the unit tests.
3. Make sure non-existent component calls are handled gracefully.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/notadamking/tensortrade/33/3548).

**Comment by gitcoinbot** [comment]
⚡️ A tip worth 0.10000  ETH (18.34 USD @ $183.38/ETH) has been granted to @mwbrulhardt for this issue from @notadamking. ⚡️ 

Nice work @mwbrulhardt! To redeem your tip, login to Gitcoin at https://gitcoin.co/explorer and select 'Claim Tip' from dropdown menu in the top right, or check your email for a link to the tip redemption page. 

 * $155944.04 in Funded OSS Work Available at: https://gitcoin.co/explorer
 * Incentivize contributions to your repo: <a href='https://gitcoin.co/tip'>Send a Tip</a> or <a href='https://gitcoin.co/funding/new'>Fund a PR</a>
 * No Email? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. Submitted 4. **Done** 

<hr>

__This Bounty has been completed.__

Additional Tips for this Bounty:
* notadamking tipped 0.1000 ETH worth 18.34 USD to mwbrulhardt.

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/notadamking/tensortrade/33/3548)
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $155,943.55 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by notadamking** [comment]
Closed by #19.
